frontend/azlinux/handle_rpm.go

```diff
@@ -87,30 +87,115 @@ func installBuildDepsPackage(target string, packageName string, w worker, deps m
+	var mounts []llb.RunOption
+	for _, data := range config.Data {
+		repoState, err := data.Spec.AsMount(data.Dest, sOpts, opts...)
+		if err != nil {
+			return nil, err
+		}
+		mounts = append(mounts, llb.AddMount(data.Dest, repoState))
+	}
+
+	return dalec.WithRunOptions(mounts...), nil
+}
+
+// meant to return a run option for importing the config files for all repos (including keys)
+func importRepos(repos []dalec.PackageRepositoryConfig, sOpt dalec.SourceOpts, opts ...llb.ConstraintsOpt) (llb.RunOption, error) {
+	configStates := []llb.RunOption{}
+	for _, repo := range repos {
+		mnts, err := importRepo(repo, sOpt, opts...)
+		if err != nil {
+			return nil, err
+		}
+
+		configStates = append(configStates, mnts...)
+	}
+
+	return dalec.WithRunOptions(configStates...), nil
+}
+
+func importRepo(config dalec.PackageRepositoryConfig, sOpt dalec.SourceOpts, opts ...llb.ConstraintsOpt) ([]llb.RunOption, error) {
+	repoConfigs := []llb.RunOption{}
+	keys := []llb.RunOption{}
+
+	for name, repoConfig := range config.Config {
+		// each of these sources represent a repo config file
+		// TODO: need to handle file vs. dir case
+		repoConfigSt, err := repoConfig.AsState(name, sOpt, opts...)
+		if err != nil {
+			return nil, err
+		}
+
+		repoConfigs = append(repoConfigs,
+			llb.AddMount(filepath.Join("/etc/yum.repos.d", name), repoConfigSt, llb.SourcePath(name)))
+	}
+
+	for name, repoKey := range config.Keys {
+		// each of these sources represent a gpg key file for a particular repo
+		gpgKey, err := repoKey.AsState(name, sOpt)
+		if err != nil {
+			return nil, err
+		}
+
+		keys = append(keys, llb.AddMount(filepath.Join("/etc/pki/rpm-gpg", name), gpgKey, llb.SourcePath(name)))
+	}
+
+	return append(repoConfigs, keys...), nil
+}
+
 func installBuildDeps(ctx context.Context, w worker, client gwclient.Client, spec *dalec.Spec, targetKey string, opts ...llb.ConstraintsOpt) (llb.StateOption, error) {
 	deps := spec.GetBuildDeps(targetKey)
 	if len(deps) == 0 {
 		return func(in llb.State) llb.State { return in }, nil
 	}
 
+	repos := spec.GetBuildRepos(targetKey)
+
 	sOpt, err := frontend.SourceOptFromClient(ctx, client)
 	if err != nil {
 		return nil, err
 	}
 
-	opts = append(opts, dalec.ProgressGroup("Install build deps"))
+	repoOpts := append(opts, dalec.ProgressGroup("Install build repos"))
+	repoMounts, err := getRepoMounts(repos, sOpt, repoOpts...)
+	if err != nil {
+		return nil, err
+	}
 
-	installOpt, err := installBuildDepsPackage(targetKey, spec.Name, w, deps, installWithConstraints(opts))(ctx, client, sOpt)
+	importRepo, err := importRepos(repos, sOpt, repoOpts...)
+	if err != nil {
+		return nil, err
+	}
+
+	opts = append(opts, dalec.ProgressGroup("Install build deps"))
+	installOpt, err := installBuildDepsPackage(targetKey, spec.Name, w, deps, importRepo, installWithConstraints(opts))(ctx, client, sOpt)
```

Comment:
the `importRepo` option which should consist of the mounts for the config files and keys is threaded through here in addition to the `Run` below. That probably isn't necessary, this was done during debugging